lizrd/core/distributed.py

```diff
@@ -0,0 +1,79 @@
+from typing import Optional, Type, Sequence
+from functools import partial
+
+from torch.distributed.fsdp.wrap import size_based_auto_wrap_policy
+from torch.distributed.fsdp import MixedPrecision, CPUOffload
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch.nn as nn
+import torch
+
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+
+def wrap_in_ddp(
+    module: nn.Module,
+    rank: int,
+):
+    return DDP(module=module.to(f"cuda:{rank}"), device_ids=[rank])
+
+
+def wrap_in_fsdp(
+    module: nn.Module,
+    rank: Optional[int],
+    param_precision: torch.dtype,
+    cast_inputs: bool,
+    mixed_precision_ignored_classes: Sequence[Type[nn.Module]],
+    offload_params: bool,
+    print_model: bool,
+    min_num_params: int,
+    modules_to_wrap: tuple[Type[nn.Module]],
+):
+    if modules_to_wrap is not None and min_num_params is not None:
+        raise Exception(
+            "The FSDP arguments `modules_to_wrap` and `min_num_params` are mutually exclusive. Either supply one, or the other."
+        )
+    if modules_to_wrap is None and min_num_params is None:
+        raise Exception(
+            "FSDP wrapping policy unspecified. Either `modules_to_wrap` or `min_num_params` must be supplied."
+        )
+
+    if modules_to_wrap is not None:
+
+        def explicit_wrap_policy(
```

Comment:
Found in documentation: ModuleWrapPolicy
https://github.com/pytorch/pytorch/blob/main/torch/distributed/fsdp/wrap.py#L185C1-L185C1

Reply:
great! Want to refactor it?